pkg/controller/cincinnati/new.go

```diff
@@ -388,3 +450,28 @@ func newCMEnvVar(name, key, cmName string) corev1.EnvVar {
 		},
 	}
 }
+
+// checksumMap produces a checksum of a ConfigMap's Data attribute. The checksum
+// can be used to detect when the contents of a ConfigMap have changed.
+func checksumMap(m map[string]string) string {
+	keys := sort.StringSlice([]string{})
+	for k := range m {
+		keys = append(keys, k)
+	}
+	keys.Sort()
+
+	hash := sha256.New()
+	encoder := base64.NewEncoder(base64.StdEncoding, hash)
+
+	// We base64 encode the data to limit the character set and then use ":" as
+	// a separator.
+	for _, k := range keys {
+		encoder.Write([]byte(k))
+		hash.Write([]byte(":"))
+		encoder.Write([]byte(m[k]))
```

Comment:
Would a Write error be problematic here?

Reply:
Yes. Oops. :innocent: Updated. Thanks for the catch!